src/Beacon.jsx

```diff
@@ -77,8 +82,28 @@ export class Beacon extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
This is an important optimization since all the states changes are causing a crazy about of re-rendering even though they often don't make any difference to the way the component is rendered (e.g. they affect how the tooltip is displayed when the beacon is clicked, but right now we are still showing the beacon).